### PR TITLE
common tests files placement refactor, moved shp.hpp up

### PR DESCRIPTION
**Pull request by lslusarczyk**
also fixed fields initialization order in dense_matrix_view
and added missing shp/examples to cmake

### CODE REVIEW

File: include/dr/shp.hpp [hunk 1]
```diff
@@ -4,10 +4,6 @@
 
 #pragma once
 
-#include <dr/detail/iterator_adaptor.hpp>
```
Comment (lslusarczyk) [inline include/dr/shp.hpp:7]: @BenBrock , please check this change. If something more should be removed or something was wrongly removed, then please write. Files here should include only direct API used by external users (no details), see this thread: https://github.com/oneapi-src/distributed-ranges/pull/190#pullrequestreview-1391885986

Comment (rscohn2) [reply include/dr/shp.hpp:7]: I probably added that before we agreed on current approach.

**Review by rscohn2 — APPROVED** [review]
Looks very good. Thanks!

File: include/dr/shp.hpp (hunk 1 above)
Comment (lslusarczyk) [reply include/dr/shp.hpp:7]: I will revert this change if we will not agree on shrinking mhp.hpp / shp.hpp by removing non-public headers.

**Comment by rscohn2** [comment]
It would be good to get this merged to avoid conflicts. If the shp examples are not easy to fix, just leave them disabled.

**Comment by lslusarczyk** [comment]
> It would be good to get this merged to avoid conflicts. If the shp examples are not easy to fix, just leave them disabled.

This is true. I've disabled them from running (some of them are meant to be run on GPU only I think), but I want still want them to be compiling in CI.

**Pull request merged**